session.go

```diff
@@ -449,6 +449,10 @@ func (s *session) retry(goCtx goctx.Context, maxCnt int, infoSchemaChanged bool)
 				break
 			}
 		}
+		if ctx.Value("mockRetryForTest") != nil {
```

Comment:
We don't need to do this.

Reply:
It's ugly but how do you test the case?

Reply:
Can we add a hook here in the future? It can be used to add more flexible tests？

Reply:
```
func retry(ctx) {
   hook := ctx.Value("mockRetryForTest")
   doActionCheckHook()
}

ctx := goctx.WithValue("mockRetryForTest", Hook)
retry(ctx)
hook.Status = 1
hook.Status = 2
hook.Status = 3
```
@zimulala